File: dotnetcoreapps/Contracts/Topic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dotnetcoreapps.Contracts
{
    public class Topic
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public TopicCategory Category { get; set; }
        public string[] Tags { get; set; }
    }
}

File: dotnetcoreapps/ClientApp/src/Entities/Topic.ts
export class Topic {
    Id: string;
    Title: string;
    Description: string;
    CreatedDate: Date;
    Category: TopicCategory;
    Tags: string[];
}

export enum TopicCategory {
    Dotnet = 0,
    DotnetCore = 1,
    AspCore = 2,
    TypeScript = 3,
    ReactJs = 4,
    Balzor = 5
}